bpf/generictracer/protocol_kafka.h

```diff
@@ -0,0 +1,523 @@
+    // we populated the hdr with message_size and correlation_id
+    // we checked also in the map of ongoing requests if there was a related
+    // ongoing request so we can hardcode the packet_type as response
+    large_buf->type = EVENT_TCP_LARGE_BUFFER;
+    large_buf->packet_type = PACKET_TYPE_RESPONSE;
+
+    large_buf->action = action;
+    large_buf->direction = direction;
+    large_buf->conn_info = pid_conn->conn;
+    large_buf->tp = req->tp;
+
+    int written = kafka_read_fixup_buffer(
+        &pid_conn->conn, large_buf->buf, &large_buf->len, u_buf, bytes_len, direction);
+    if (written < 0) {
+        bpf_dbg_printk("kafka_send_large_buffer: failed to read buffer, not sending large buffer");
+        return 0;
+    }
+
+    u32 total_size = sizeof(tcp_large_buffer_t);
+    total_size += written > sizeof(void *) ? written : sizeof(void *);
+
+    req->has_large_buffers = true;
+    bpf_ringbuf_output(&events, large_buf, total_size & k_large_buf_max_size_mask, get_flags());
+
+    bpf_dbg_printk(
+        "kafka_send_large_buffer: sent large buffer bytes_len %d, packet_type %d, direction %d, "
+        "message_size %d, correlation_id %d, total_size %d, total_size & k_large_buf_max_size_mask "
+        "%d",
+        bytes_len,
+        PACKET_TYPE_RESPONSE,
+        direction,
+        hdr.message_size,
+        hdr.correlation_id,
+        total_size,
+        total_size & k_large_buf_max_size_mask);
+    return 0;
+}
+
+static __always_inline u8 is_kafka(connection_info_t *conn_info,
+                                   const unsigned char *data,
+                                   u32 data_len,
+                                   enum protocol_type *protocol_type,
+                                   u8 direction) {
+    if (*protocol_type != k_protocol_type_kafka && *protocol_type != k_protocol_type_unknown) {
+        // Already classified, not kafka.
+        return 0;
+    }
+
+    if (kafka_store_state_data(conn_info, data, (size_t)data_len, direction) < 0) {
+        bpf_dbg_printk("is_kafka: 4 bytes packet, storing state data");
+        return 0;
+    }
+
+    struct kafka_request_hdr req_hdr = {};
+    struct kafka_response_hdr res_hdr = {};
+    if (kafka_parse_fixup_request_header(conn_info, &req_hdr, data, data_len, direction) == 0) {
+        if (req_hdr.message_size > k_kafka_message_size_max) {
+            bpf_dbg_printk("is_kafka: request message size is too large: %d", req_hdr.message_size);
+            return 0;
+        }
+        struct kafka_correlation_data correlation_data = {};
+        correlation_data.correlation_id = req_hdr.correlation_id;
+        bpf_map_update_elem(&kafka_ongoing_requests, conn_info, &correlation_data, BPF_ANY);
+        bpf_dbg_printk("is_kafka: kafka! message_size %d, request_api_key %d",
+                       req_hdr.message_size,
+                       req_hdr.request_api_key);
+        bpf_dbg_printk("is_kafka: kafka! request_api_version %d, correlation_id=%d",
```

Comment:
I think it's too verbose, one message is enough, maybe just put api key and correlation id